plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/KubernetesClusterHelperImpl.java

```diff
@@ -58,6 +58,9 @@ public void checkVmCanBeDestroyed(UserVm userVm) {
         KubernetesCluster kubernetesCluster = kubernetesClusterDao.findById(vmMapVO.getClusterId());
         String msg = "Instance is a part of a Kubernetes cluster";
         if (kubernetesCluster != null) {
+            if (kubernetesCluster.getClusterType().equals(KubernetesCluster.ClusterType.ExternalManaged)) {
```

Comment:
```suggestion
            if (KubernetesCluster.ClusterType.ExternalManaged.equals(kubernetesCluster.getClusterType())) {
```
check should be safer other way around